theano/sandbox/rng_mrg.py

```diff
@@ -1551,6 +1552,17 @@ def normal(self, size, avg=0.0, std=1.0, ndim=None,
         return final_samples
 
 
+@register_opt2([mrg_uniform], 'fast_compile')
+def local_gpua_mrg1(op, context_name, inputs, outputs):
+    if (type(op) == mrg_uniform and
+            isinstance(inputs[0].type, GpuArrayType)):
+        outs = GPUA_mrg_uniform.new(inputs[0],
+                                    op.output_type.ndim,
+                                    op.output_type.dtype,
+                                    inputs[1])
+        return [outs[0], host_from_gpua(outs[1])]
```

Comment:
Yeah, will make a call to the existing optimization ( I remember addressing as well. Don't know how i lost track.